### ISSUE TITLE
HTTP API race condition

### ISSUE THREAD

**Issue by codingberlin** [orientechnologies/orientdb#6601]
## OrientDB Version, operating system, or hardware.
- problem occurs since v2.2.4
## Operating System
- [X] Linux
- [X] MacOSX
## Expected behavior and actual behavior

Expected:
When using the Http API and POST a batch command there should only be one response.

Actual:
There are two responses. First is 200, second is 204.
## Steps to reproduce the problem

If you checkout https://github.com/ImmobilienScout24/play-orientdb-client/tree/orientdb2.2.4 and run the tests with `./sbt test` while sniffing the network traffic, the result looks like this:

```
POST /batch/temp HTTP/1.1
Host: localhost:2480
Authorization: Basic cm9vdDpyb290
User-Agent: akka-http/2.4.7
Content-Type: application/json
Content-Length: 230

{"transaction":true,"operations":[{"language":"sql","script":["Update SchemaLock set lockedBy = \"isdeblnnm150.iscout.local\" where (lockedBy is null or lockedBy = \"isdeblnnm150.iscout.local\") and id=\"did\""],"type":"script"}]}HTTP/1.1 200 OK
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Date: Wed Aug 24 13:13:11 CEST 2016
Content-Type: application/json; charset=utf-8
Server: OrientDB Server v.2.2.4
Connection: Keep-Alive
Content-Transfer-Encoding: binary
Transfer-Encoding: chunked

31
{"result":[{"@type":"d","@version":0,"value":1}]}
0

HTTP/1.1 204 OK
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Date: Wed Aug 24 13:13:11 CEST 2016
Content-Type: text/plain; charset=utf-8
Server: OrientDB Server v.2.2.4
Connection: Keep-Alive
Set-Cookie: OSESSIONID=OS1472037191295-2818016975438360849; Path=/; HttpOnly
Content-Length: 0
```

As you can see there are two responses in the tcp stream. The first is the correct 200. The second is 204 which causes our tests to break.

Maybe an important hint is that the connection is kept alive.

With version 2.2.3 ( https://github.com/ImmobilienScout24/play-orientdb-client ) the problem does not occur.

**Comment by tglman** [comment]
hi @codingberlin,

Thank you for the detailed report, we are going to check as soon as possible and let you know.

Regards

**Comment by maggiolo00** [comment]
@codingberlin 

i think the problem is the streaming response.

Meanwhile you can disable that in the config.xml of orientdb and it should works as expected

```
<entry name="network.http.streaming" value="false" ></entry>
```

Thanks

Enrico

**Comment by codingberlin** [comment]
@maggiolo00 
Thank you for the workaround. It will help others who run into the same problem.

Unfortunately our project is an OrientDB client for the Scala language thus we have no access to the OrientDB configuration of the users of our client.

Regards

**Comment by tglman** [comment]
hi @codingberlin,

We refactored the streaming activation in 2.2.10 this error should not happen anymore, closing.

**Comment by codingberlin** [comment]
Unfortunately the issue still occurs.

I updated orientdb to 2.2.11 in a new branch:
https://github.com/ImmobilienScout24/play-orientdb-client/tree/orientdb2.2.11

Then I ran ```./sbt "testOnly de.is24.play.orientdb.EmbeddedOrientDBTest"``` while sniffing the network traffic.

The result is rather worse:
```
command/temp/sql HTTP/1.1
Host: localhost:2480
Authorization: Basic cm9vdDpyb290
User-Agent: akka-http/2.4.7
Content-Type: text/plain; charset=UTF-8
Content-Length: 95

select count(*) from (select expand(classes) from metadata:schema) where name = "SchemaVersion"HTTP/1.1 200 OK
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Date: Di, 04 Okt 2016 08:25:34 UTC
Content-Type: application/json; charset=utf-8
Server: OrientDB Server v.2.2.11
Connection: Keep-Alive
Content-Transfer-Encoding: binary
Transfer-Encoding: chunked

58
{"result":[{"@type":"d","@rid":"#-2:0","@version":0,"count":0,"@fieldTypes":"count=l"}]}
HTTP/1.1 200 OK
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Date: Di, 04 Okt 2016 08:25:34 UTC
Content-Type: application/json; charset=utf-8
Server: OrientDB Server v.2.2.11
Connection: Keep-Alive
Content-Transfer-Encoding: binary
Transfer-Encoding: chunked

58
{"result":[{"@type":"d","@rid":"#-2:0","@version":0,"count":0,"@fieldTypes":"count=l"}]}
0

POST /batch/temp HTTP/1.1
Host: localhost:2480
Authorization: Basic cm9vdDpyb290
User-Agent: akka-http/2.4.7
Content-Type: application/json
Content-Length: 255

{"transaction":false,"operations":[{"language":"sql","script":["Create class SchemaLock","Create property SchemaLock.id String","Create index idIndex on SchemaLock (id) UNIQUE","Insert into SchemaLock set id = \"did\", lockedBy = null"],"type":"script"}]}HTTP/1.1 200 OK
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Date: Di, 04 Okt 2016 08:25:34 UTC
Content-Type: application/json; charset=utf-8
Server: OrientDB Server v.2.2.11
Connection: Keep-Alive
Content-Transfer-Encoding: binary
Transfer-Encoding: chunked

67
{"result":[{"@type":"d","@rid":"#25:0","@version":1,"@class":"SchemaLock","lockedBy":null,"id":"did"}]}
HTTP/1.1 200 OK
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Date: Di, 04 Okt 2016 08:25:34 UTC
Content-Type: application/json; charset=utf-8
Server: OrientDB Server v.2.2.11
Connection: Keep-Alive
Content-Transfer-Encoding: binary
Transfer-Encoding: chunked

67
{"result":[{"@type":"d","@rid":"#25:0","@version":1,"@class":"SchemaLock","lockedBy":null,"id":"did"}]}
0

HTTP/1.1 204 OK
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Date: Di, 04 Okt 2016 08:25:34 UTC
Content-Type: text/plain; charset=utf-8
Server: OrientDB Server v.2.2.11
Connection: Keep-Alive
Set-Cookie: OSESSIONID=OS1475569534826-6549190054912557330; Path=/; HttpOnly
Content-Length: 0
```
In the sniff above there are two requests.
Both were respond by duplicate responses while the first missed the finishing ```0``` (chunking end).
The second request is answered with three responses. The last is 204.

Could you take another look at this?

**Comment by tglman** [comment]
hi @codingberlin,

We did some fix on this topic in 2.2.14, can you double check if this case is fixed>

Regards

**Comment by lvca** [comment]
Closed because no response.
